src/pdm/resolver/providers.py

```diff
@@ -188,11 +188,19 @@ def _find_candidates(self, requirement: Requirement) -> Iterable[Candidate]:
                             if parsed_version.is_prerelease:
                                 prerelease = True
                                 break
-            return self.repository.find_candidates(
-                requirement,
-                self.allow_prereleases if prerelease is None else prerelease,
-                minimal_version=self.direct_minimal_versions and self._is_direct_requirement(requirement),
+            candidates = list(
+                self.repository.find_candidates(
+                    requirement,
+                    self.allow_prereleases if prerelease is None else prerelease,
+                    minimal_version=self.direct_minimal_versions and self._is_direct_requirement(requirement),
+                )
             )
+            wheel_candidates = [c for c in candidates if c.link and c.link.is_wheel]
+            if wheel_candidates:
+                # Return wheels only for resolution to speed up.
+                # When none of the wheels satisfies, no need to try the sdists.
```

Comment:
none -> one?

Reply:
No, it's correct. In other words, we only need to evaluate the wheels. If none of them are satisfied, the entire version can be skipped without considering sdist.